### PR TITLE
1267: Event detail page design refresh

### PR DESCRIPTION
**Pull request by stevejalim**
This changeset adds a refresh of the Event page.

(Related issue #1267)

## Key changes:

- Data model and Wagtail editing UI updated to hide fields we no longer need and add two new ones:
  - link to 'event content'
  - link to 'official website'

- New design laid in, including
    - header w with Event icon
    - switching to main + sidebar layout
    - new buttons to link out to relevant pages, which appear only when configured
    - sidebar panel shows a card for the same event, and links out to the "most appropriate" thing, which would be the first available link from: event content, then event official site, then registration page. This is better than the card linking to the page it is on.

- Checked on desktop and mobile layouts
- HTML valid 

## How to test

- The code is up on dev and I've made a sample page: https://developer-portal.dev.mdn.mozit.cloud/events/test-event/

(We should do a cross-browser sweep once all the pages are in, but any cross-browser checks you can do are very welcome right now)

## Screenshots

Screenshot_2020-03-27 event x(1)

Screenshot_2020-03-27 event x

### CODE REVIEW

**Comment by schalkneethling** [comment]
I wonder if it is worth showing the icon on mobile in the header:

Screenshot 2020-03-30 at 11 21 03

See below the layout without the icon

Screenshot 2020-03-30 at 11 22 25

**Comment by schalkneethling** [comment]
On iPad in portrait mode (768x1024) the buttons and share component alignment seems incorrect:

Screenshot 2020-03-30 at 11 23 43

**Comment by schalkneethling** [comment]
Also on mobile, I wonder if these buttons:

Screenshot 2020-03-30 at 11 25 27

Should not be stack as below?

Screenshot 2020-03-30 at 11 25 52

Setting `display: block;` on this anchor elements will do the trick.

File: developerportal/apps/events/templates/event.html [hunk 1]
```diff
@@ -1,57 +1,73 @@
...
+        {% if page.official_website %}
+          <a href="{{page.register_url}}" class="mzp-c-button">
+            Official Website
+          </a>
+        {% endif %}
+        {% if page.event_content %}
+          <a href="{{page.register_url}}" class="mzp-c-button">
+            Event Content
+          </a>
+        {% endif %}
```
Comment (schalkneethling) [inline developerportal/apps/events/templates/event.html]: Seems like this is a candidate for an unordered list?

Comment (stevejalim) [reply developerportal/apps/events/templates/event.html]: Good call - will make the change as part of a mop-up ticket, if that's ok. Noted in #1233

**Comment by valgrimm** [comment]
@stevejalim Proposal: On mobile, pop the calendar icon in above the title, to be consistent with the way the icon for  video is handled. Just noticed this, partially much thanks to schalk.

**Comment by stevejalim** [comment]
@valgrimm Actually, I was about to say that I think we should drop it, to avoid making the header unnecessarily tall (ie, go with what Schalk is suggesting). 

(We'd do the same for all mobile headers, then)

Can i steer you the other way? 😄

File: developerportal/apps/events/templates/event.html (hunk 1 above)
Comment (stevejalim) [reply developerportal/apps/events/templates/event.html]: Thanks for the pointer about EOL @schalkneethling! Noted in the mop-up ticket

**Comment by valgrimm** [comment]
@stevejalim For now, yes, drop the icons on mobile. Handling all of the icons responsively will add overhead we cannot afford right now. I'll add a ticket to handle this after the redesign is rolled out. As long as the solution to these icons is handled consistently for all pages, I am good with it. #1323 is the ticket for making these icons responsive @malqinneh This is a short-term solution to reduce scope.

File: developerportal/apps/events/templates/event.html (hunk 1 above)
Comment (stevejalim) [reply developerportal/apps/events/templates/event.html]: Done!

**Pull request merged**